packages/reactivity/src/watch.ts

```diff
@@ -215,8 +215,12 @@ export function watch(
     effect.stop()
     if (scope) {
       remove(scope.effects, effect)
+      remove(scope.cleanups, watchHandle)
     }
   }
+  if (scope) {
+    scope.cleanups.push(watchHandle)
+  }
```

Comment:
Just a nit, could replace with `onScopeDispose(watchHandle)`

Reply:
This will cause an additional check for `if(activeEffectScope){}`. It will generate a warning if we don't include `if(scope){}` and directly call `onScopeDispose`.

Reply:
Updated